easybuild/framework/easyblock.py

```diff
@@ -914,7 +914,9 @@ def make_module_extend_modpath(self):
             modpath_exts = ActiveMNS().det_modpath_extensions(self.cfg)
             self.log.debug("Including module path extensions returned by module naming scheme: %s" % modpath_exts)
             full_path_modpath_extensions = [os.path.join(top_modpath, mod_path_suffix, ext) for ext in modpath_exts]
-            # module path extensions must exist, otherwise loading this module file will fail
+            # module path extensions must exist since the generated module will run 'module use' on them;
+            # this is only really true for Tcl/C env mods, but we cannot make any assumptions on which modules tool
+            # will be used to consume the generated module (better safe than sorry)
             for modpath_extension in full_path_modpath_extensions:
```

Comment:
and/or: include a guard `file exists $dir` around `module use` statements in Tcl module files, to avoid Tcl/C env mods barfing when a `module use` is run on a non-existing directory

bonus points: optimise by only running `module use` on existing *non-empty* directories (but the question is then whether there is an (efficient) way of checking if a directory is empty or not)